### PR TITLE
Adobe Analytics integration v1

### PR DESCRIPTION
**Pull request by tjleing**
- Adds Shortbread integration for EU user cookie consent
- Adds Adobe Analytics integration
  - page load event (seems to come with referrer, so we don't need to set that up manually?)
  - external link event (also done by default)
  - docsearch events -- including search query and result count

Theoretically TODO: internal/external campaign query search parameter
Will be able to see analytics data, including hopefully a Re:Invent bump, only after these changes are merged in

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jamesonwilliams — COMMENTED** [review]
`s.eVar26`, `s.eVar27`, `s.tl`, `s.t` -- very cryptic. Are these minified references, or something?

**Comment by tjleing** [comment]
> `s.eVar26`, `s.eVar27`, `s.tl`, `s.t` -- very cryptic. Are these minified references, or something?

Unfortunately we can't change them -- they're required by Adobe Analytics!  Not entirely sure why it's so cryptic except possibly to cut down on script and query string parameter length.

**Review by mauerbac — APPROVED** [review]
LGTM - @sammartinez / @harrysolovay , can you also review ?

File: client/src/docs-ui/page/page.tsx [hunk 1]
```diff
@@ -26,7 +26,12 @@ import {
   SetNewSelectedTabHeadings,
 } from "./page.types";
 import {pageContext} from "./page.context";
-import {track, AnalyticsEventType} from "../../utils/track";
+import {
+  track,
```
Comment (sammartinez) [inline client/src/docs-ui/page/page.tsx:30]: So I see `trackPageVisit()` and `trackPageFetchException()` in use, where is `track` or the `AnalyticsEventType` used?

File: client/src/utils/track.ts [hunk 2]
```diff
@@ -7,6 +7,14 @@ let configured = false;
 if (!configured) {
   Auth.configure(awsexports);
   Analytics.configure(awsexports);
+  // @ts-ignore
+  if (typeof AWSCShortbread !== "undefined") {
```
Comment (sammartinez) [inline client/src/utils/track.ts:11]: Is there an additional check that we want to place here? Does it turn into a function? Does it need to be: 
```suggestion
  if (typeof AWSCShortbread !== undefined) {
```

File: client/src/utils/track.ts [hunk 3]
```diff
@@ -7,6 +7,14 @@ let configured = false;
...
+    // @ts-ignore
+    const shortbread = AWSCShortbread({
+      domain: ".amplify.aws",
+    });
+    shortbread.checkForCookieConsent();
```
Comment (sammartinez) [inline client/src/utils/track.ts:16]: Is shortbread used anywhere else within this file?

File: client/src/utils/track.ts [hunk 4]
```diff
@@ -7,6 +7,14 @@ let configured = false;
 if (!configured) {
   Auth.configure(awsexports);
   Analytics.configure(awsexports);
+  // @ts-ignore
+  if (typeof AWSCShortbread !== "undefined") {
+    // @ts-ignore
+    const shortbread = AWSCShortbread({
```
Comment (sammartinez) [inline client/src/utils/track.ts:13]: Do we need to create a new instance as follows? 
```suggestion
    const shortbread = new AWSCShortbread({
```

File: client/src/utils/track.ts [hunk 5]
```diff
@@ -65,3 +73,35 @@ export const track = (event: AnalyticsEvent): Promise<unknown> | undefined => {
     }
   }
 };
+
+export const trackPageVisit = (): void => {
+  if (Build.isBrowser) {
+    // @ts-ignore
+    s.t();
```
Comment (sammartinez) [inline client/src/utils/track.ts:80]: We can talk offline but what does this do?

File: client/src/utils/track.ts [hunk 6]
```diff
@@ -65,3 +73,35 @@ export const track = (event: AnalyticsEvent): Promise<unknown> | undefined => {
...
+    s.eVar26 = query;
+  }
+};
+
+export const trackSearchResult = (resultCount: number): void => {
+  if (Build.isBrowser) {
+    // @ts-ignore
+    s.eVar27 = resultCount;
+    // @ts-ignore
+    s.events = resultCount === 0 ? "event1" : "event2";
```
Comment (sammartinez) [inline client/src/utils/track.ts:103]: what do these events mean?

File: docs/scripts/shortbreadv1.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
...x-c #awsccc-sb-a.awsccc-Rtl #awsccc-cb-c #awsccc-cb-title{padding-right:40px}}#awsccc-sb-ux-c #awsccc-sb-a a,#awsccc-sb-ux-c #awsccc-sb-a a>span,#awsccc-sb-ux-c #awsccc-sb-a a svg path{color:#0073bb;text-decoration:none}#awsccc-sb-ux-c #awsccc-sb-a a:hover,#awsccc-sb-ux-c #awsccc-sb-a a>span:hover,#awsccc-sb-ux-c #awsccc-sb-a a svg path:hover{color:#0073bb;text-decoration:underline}#awsccc-sb-ux-c #awsccc-sb-a .awsccc-tab-helper{outline:0;text-decoration:none}.awsccc-cs-modal-open{overflow:hidden;-webkit-box-sizing:border-box;box-sizing:border-box}',""]),e.exports=a},function(e,a,t){"use strict";Object.defineProperty(a,"__esModule",{value:!0}),a.convertToArray=a.update=void 0,a.update=function(e,a){return Object.keys(a).forEach((function(t){e[t]=a[t]})),e},a.convertToArray=function(e){return Array.prototype.slice.call(e)}},function(e,a,t){"use strict";Object.defineProperty(a,"__esModule",{value:!0});a.default=function(e,a,t){function n(){a.removeEventListener("DOMContentLoaded",n),t.removeEventListener("load",n),e()}void 0===a&&(a=document),void 0===t&&(t=window),"loading"!==a.readyState?t.setTimeout(e):(a.addEventListener("DOMContentLoaded",n),t.addEventListener("load",n))}}])}));
```
Comment (sammartinez) [inline docs/scripts/shortbreadv1.js:1]: Lets chat about this offline as well.

File: client/src/docs-ui/page/page.tsx (hunk 1 above)
Comment (tjleing) [reply client/src/docs-ui/page/page.tsx:30]: `track` and `AnalyticsEventType` are used right above the `trackPageVisit` and `trackPageFetchException` calls -- they're being used for Pinpoint tracking right now.

Comment (tjleing) [reply client/src/docs-ui/page/page.tsx:30]: We'll rip out those `track()` calls once we're fully switched over to Adobe from Pinpoint.  I'm still unsure whether a `switch`-based approach (i.e. calling `track()` with a given event type) or this current approach with the specific named methods like `trackPageVisit` is more appropriate

File: client/src/utils/track.ts (hunk 2 above)
Comment (tjleing) [reply client/src/utils/track.ts:11]: I tried that first and it doesn't work... actually it might be better to use `if (Build.isBrowser)` instead as is done in `track.ts` -- the code is there fully for testing

File: client/src/utils/track.ts (hunk 3 above)
Comment (tjleing) [reply client/src/utils/track.ts:16]: Think that's it -- Shortbread does a lot automatically

File: client/src/utils/track.ts (hunk 4 above)
Comment (tjleing) [reply client/src/utils/track.ts:13]: Strangely no, it's `AWSCShortbread({...})` -- not sure why 😅

File: client/src/utils/track.ts (hunk 3 above)
Comment (sammartinez) [reply client/src/utils/track.ts:16]: can you shorten this to be a one line then? 
```suggestion
    AWSCShortbread({ domain: ".amplify.aws" }).checkForCookieConsent();
```

File: client/src/utils/track.ts (hunk 2 above)
Comment (sammartinez) [reply client/src/utils/track.ts:11]: Interesting why that wouldn't work 🤔

**Review by harrysolovay — APPROVED** [review]
LGTM

File: client/src/utils/track.ts (hunk 3 above)
Comment (tjleing) [reply client/src/utils/track.ts:15]: ooh probably.. not sure why I didn't see this before

Comment (tjleing) [reply client/src/utils/track.ts]: fixed!
